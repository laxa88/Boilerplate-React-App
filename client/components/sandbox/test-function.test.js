import React from "react";
import * as enzyme from "enzyme";
import TestFunction from "./test-function";

describe("test-function.js", () => {
  it("Should throw warning if no props passed in.", () => {
    const spy = jest.spyOn(global.console, "error").mockImplementation(() => null);

    enzyme.shallow(<TestFunction />);
    expect(spy).toBeCalled();

    spy.mockReset();
  });

  it("Should mount.", () => {
    const wrapper = enzyme.mount(<TestFunction customHandler={() => { console.log("WOAH"); }} />);
    expect(wrapper).toBeDefined();
  });

  it("Should call the custom callback function that was passed in as prop.", () => {
    const spy = jest.fn();

    const wrapper = enzyme.mount(<TestFunction customHandler={spy} />);
    expect(wrapper).toBeDefined();

    const addButton = wrapper.find("#callback-button");
    addButton.simulate("click");
    expect(spy).toHaveBeenCalled();
  });
});
