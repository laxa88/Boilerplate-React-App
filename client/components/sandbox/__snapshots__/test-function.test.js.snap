// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-function.js Should shallow render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <p>
      <button
        id="callback-button"
        onClick={[Function]}
      >
        CUSTOM HANDLER
      </button>
    </p>
  </div>,
  "nodes": Array [
    <div>
      <p>
        <button
          id="callback-button"
          onClick={[Function]}
        >
          CUSTOM HANDLER
        </button>
      </p>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TestFunction
        customHandler={[Function]}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TestFunction {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "customHandler": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <p>
            <button
              id="callback-button"
              onClick={[Function]}
            >
              CUSTOM HANDLER
            </button>
          </p>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <p>
            <button
              id="callback-button"
              onClick={[Function]}
            >
              CUSTOM HANDLER
            </button>
          </p>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TestFunction
    customHandler={[Function]}
  />,
}
`;
